packages/checkout-ui-extensions/src/style/style.ts

```diff
@@ -0,0 +1,66 @@
+import {defaultMemoize as memoize} from 'reselect';
```

Comment:
**TLDR;** This file will live in `checkout-web/packages/checkout-ui-extensions/documentation/conditional-props.md` and should be manually updated if anything changes here. 

Additional Context:
The Style implementation doc will actually live in /packages/checkout-ui-extensions/documentation/conditional-props.md within the checkout-web repo. When new versions of the package are published, that entire package is copied over to ui-extensions and then a yarn docs:checkout script is run which generates Readme files for components and all the appropriate files for shopify-dev.

And the reason why we had to use a static markdown file for this - is because the doc parser @shopify/docs-tools relies on interfaces as the entry point to document things. So, it has a lot of trouble with things like React Hooks and these types of functions such as the Style Helper.

While it would have been nice for us to update docs-tools to support this its currently an unsupported package we are stuck using and is really difficult to work on. I’ve made some updates to it, but its a beast and frankily a bit of a mess. So, this is the workaround we’ll need to go with for Editions.

The Core build team is working on a new shopify-wide way of documenting typescript projects https://vault.shopify.io/projects/24049. We can move to that eventually :crossed_fingers: